processor/src/lib.rs

```diff
@@ -41,21 +40,23 @@ impl Process {
         return index;
     }
 
-    pub fn get_index_value(&self, op_str: &str)-> u32 {
+    pub fn get_index_value(&self, op_str: &str)-> (u32, ImmediateOrRegName) {
         let src = op_str.parse();
         let mut value = Default::default();
         if src.is_ok() {
             value = src.unwrap();
+            return (value, ImmediateOrRegName::Immediate(Felt(value)));
         } else {
             let src_index = self.get_reg_index(op_str);
             value = self.registers[src_index];
+            return (value, ImmediateOrRegName::RegName(src_index as u8));
         }
-        value
+
     }
 
-    pub fn execute(&mut self, program: &Program) -> Result<(), String> {
+    pub fn execute(&mut self, program: &mut Program) -> Result<(), String> {
```

Comment:
seperate trace from program, and return trace maybe better?

Reply:
Executor execute program in binary form and decode happens here makes more intuitive sense. And pc can increase depending on instruction size which could be a function defined in every instruction.

Reply:
Another suggestion, unwrap should not used here and there, error should be handled :P

Reply:
some unwrap is made sure not trigger err.so use it